cmake/OpenCVDetectWebNN.cmake

```diff
@@ -0,0 +1,22 @@
+ocv_clear_vars(HAVE_WEBNN)
+ocv_clear_vars(WEBNN_EMSDK)
+if(WITH_WEBNN)
+  set(WEBNN_HEADER_DIRS "$ENV{WEBNN_NATIVE_DIR}/out/Release/gen/src/include")
```

Comment:
Please require developer to set `WEBNN_NATIVE_DIR` including the output directory, say including the `out/Release`. Because developer may build webnn-native in different output directory, we should not hard code it here.

Reply:
Hi Ningxin, the environment variable `WEBNN_NATIVE_DIR` is used in two places:
1. `set(WEBNN_HEADER_DIRS "$ENV{WEBNN_NATIVE_DIR}/out/Release/gen/src/include")`
2. `set(WEBNN_INCLUDE_DIRS "$ENV{WEBNN_NATIVE_DIR}/src/include")`

If `out/Release` is included in `WEBNN_NATIVE_DIR`, the second place will need another environment variable.

Reply:
You can set it as `$ENV{WEBNN_NATIVE_DIR}/../../src/include`. It assumes the two hierarchies, like `out/Release`, however I think it is probably a most common case.

Reply:
Thanks! I previously thought that CMake cannot use such method. I've updated this.